File: src/main/java/com/pickme/calendar/entity/Calendar.java
package com.pickme.calendar.entity;

import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.lang.NonNull;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Document(collection = "calendar")
@Getter
@RequiredArgsConstructor
public class Calendar {

	@Id
	private String id;

	@NonNull
	private String clientId;

	@NonNull
	private List<InterviewDetail> interviewDetails;
}

File: src/main/kotlin/com/pickme/calendar/entity/Calendar.kt
package com.pickme.calendar.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "calendar")
data class Calendar(
    @Id
    val id: String = org.bson.types.ObjectId().toString(),

    val clientId: String,

    var interviewDetails: MutableList<InterviewDetail> = mutableListOf(),
)
